syscalls.c

```diff
@@ -1281,12 +1545,33 @@ ssize_t syscalls_sys_recvmsg(char *ustack)
 	GETFROMSTACK(ustack, struct msghdr *, msg, 1);
 	GETFROMSTACK(ustack, int, flags, 2);
 
+	if (vm_mapBelongs(proc, msg, sizeof(*msg)) < 0) {
+		return -EFAULT;
+	}
+
+	if (vm_mapBelongs(proc, msg->msg_iov, sizeof(*msg->msg_iov) * msg->msg_iovlen) < 0) {
+		return -EFAULT;
+	}
+
+	if ((msg->msg_iov->iov_base != NULL) && (vm_mapBelongs(proc, msg->msg_iov->iov_base, msg->msg_iov->iov_len) < 0)) {
```

Comment:
This should be a loop over `msg->msg_iov` with `msg->msg_iovlen` iterations (although we don't support `iovlen` > 1 it yet)